src/api/mutant/Mutator.ts

```diff
@@ -14,9 +14,10 @@ interface Mutator {
    * If the mutator wants to mutate the node, it should return a clone of the node with mutations,
    * otherwise null. 
    * @param node A FROZEN Node which could be cloned and mutated.
+   * @param deepCopy A function to create a deep copy of an object.
    * @returns An array of mutated Nodes.
    */
-  applyMutations(node: ESTree.Node): ESTree.Node[];
+  applyMutations(node: ESTree.Node, deepCopy: (obj: any) => any): ESTree.Node[];
```

Comment:
As an aside: I think a shallow copy is enough. You generally only want to alter the current node anyway. The children can still point to the frozen nodes. Maybe we can add a copy function with a `boolean` parameter (`deep`).